### PR TITLE
TV Android Sample App: Dialogs for new commissioning feedback and message cluster

### PR DESCRIPTION
**Pull request by chrisdecenzo**
Add pop-up dialogs for new 1.3 features
- Commissioning dialogs for errors and commissioner passcode
- Message display dialogs

### CODE REVIEW

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 1]
```diff
@@ -158,8 +181,37 @@ void JNIMyUserPrompter::PromptForCommissionPasscode(uint16_t vendorId, uint16_t
...
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompter-JNI.cpp:196]: please try not to use utfstring that created extra globalref, you can use env->NewStringUTF(data). I will try to find sometime to cleanup UtfString in later weeks, thanks

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 2]
```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
...
  */
 void JNIMyUserPrompter::PromptCommissioningStarted(uint16_t vendorId, uint16_t productId, const char * commissioneeName)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::PromptCommissioningStarted Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::PromptCommissioningStarted");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompter-JNI.cpp:281]: check env nullptr, thanks

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 3]
```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
...
+    std::string stringCommissioneeName(commissioneeName);
+    std::string stringPairingInstruction(pairingInstruction);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompter-JNI.cpp:249]: same as the above, use env->NewStringUTF, thanks

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 4]
```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
...
+        UtfString jniPairingInstruction(env, stringPairingInstruction.data());
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompter-JNI.cpp:250]: same as the above, use env->NewStringUTF, thanks

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 5]
```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
...
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+    std::string stringCommissioneeName(commissioneeName);
+
+    VerifyOrExit(mJNIMyUserPrompterObject.HasValidObjectRef(), err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mPromptForCommissionOKPermissionMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != nullptr, err = CHIP_JNI_ERROR_NO_ENV);
+
+    {
+        UtfString jniCommissioneeName(env, stringCommissioneeName.data());
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompter-JNI.cpp:289]: same as the above, use env->NewStringUTF, thanks

File: examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp [hunk 6]
```diff
@@ -64,3 +64,38 @@ JNI_METHOD(void, OnPromptDeclined)(JNIEnv *, jobject)
...
+    chip::DeviceLayer::StackLock lock;
+    ChipLogProgress(Zcl, "OnCommissionerPasscodeCancel");
+    GetCommissionerDiscoveryController()->Cancel();
+#endif
+}
+
+JNI_METHOD(void, OnMessageResponse)(JNIEnv * env, jobject, jstring jMessageId, jlong jOptionId)
+{
+    chip::DeviceLayer::StackLock lock;
+    uint64_t optionid = (uint64_t) jOptionId;
```
Comment (yunhanw-google) [inline examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp:89]: static_cast?

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp (hunk 1 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompter-JNI.cpp]: addressed

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp (hunk 3 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompter-JNI.cpp]: addressed

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp (hunk 5 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompter-JNI.cpp]: addressed

File: examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp (hunk 6 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp]: addressed

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp [hunk 7]
```diff
@@ -182,17 +233,78 @@ bool JNIMyUserPrompter::DisplaysPasscodeAndQRCode()
 void JNIMyUserPrompter::PromptWithCommissionerPasscode(uint16_t vendorId, uint16_t productId, const char * commissioneeName,
                                                        uint32_t passcode, uint16_t pairingHint, const char * pairingInstruction)
 {
-    // TODO
-    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode Needs Implementation");
+    ChipLogError(Zcl, "JNIMyUserPrompter::PromptWithCommissionerPasscode");
+
+    DeviceLayer::StackUnlock unlock;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
```
[the remark this answers is not in the document]
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompter-JNI.cpp:243]: its 4 lines below

File: examples/tv-app/android/java/MyUserPrompter-JNI.cpp (hunk 2 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompter-JNI.cpp:286]: its 4 lines below

**Review by yunhanw-google — APPROVED** [review]

**Review by sharadb-amazon — APPROVED** [review]

File: examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp [hunk 8]
```diff
@@ -64,3 +64,38 @@ JNI_METHOD(void, OnPromptDeclined)(JNIEnv *, jobject)
     GetCommissionerDiscoveryController()->Cancel();
 #endif
 }
+
+JNI_METHOD(void, OnCommissionerPasscodeOK)(JNIEnv *, jobject)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
+    chip::DeviceLayer::StackLock lock;
+    ChipLogProgress(Zcl, "OnCommissionerPasscodeOK");
+    // GetCommissionerDiscoveryController()->Ok();
```
Comment (lazarkov) [inline examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp:73]: Commented out by mistake?

**Review by lazarkov — APPROVED** [review]

File: examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp (hunk 8 above)
Comment (chrisdecenzo) [reply examples/tv-app/android/java/MyUserPrompterResolver-JNI.cpp:73]: there is no callback right now to its more of a TODO

**Pull request merged**